index.bs

```diff
@@ -2730,7 +2730,7 @@ partial interface MLGraphBuilder {
     The <dfn method for=MLGraphBuilder>gather(|input|, |indices|, |options|)</dfn> method steps are:
   </summary>
     1. If [=MLGraphBuilder/validating operand=] with [=this=] and any of |input| and |indices| returns false, then [=exception/throw=] a {{TypeError}}.
-    1. If |indices|'s [=MLOperand/dataType=] is neither {{MLOperandDataType/"uint32"}} nor {{MLOperandDataType/"int64"}}, then [=exception/throw=] a {{TypeError}}.
+    1. If |indices|'s [=MLOperand/dataType=] is not {{MLOperandDataType/"uint32"}} or {{MLOperandDataType/"int64"}}, then [=exception/throw=] a {{TypeError}}.
```

Comment:
Hmm, I found `neither nor` less ambiguous than `not x or y` (is that `(not x) or y` or `not (x or y)`?...). Yes, it's a little longer, but clearer imo. Propose reverting:

```suggestion
    1. If |indices|'s [=MLOperand/dataType=] is neither {{MLOperandDataType/"uint32"}} nor {{MLOperandDataType/"int64"}}, then [=exception/throw=] a {{TypeError}}.
```

Reply:
Here are sample cases from the follow-on change, and why I shied away from neither/nor:

* If input’s dataType is not "float32" or "float16", then ...
* If input’s dataType is not "float32", "float16", "int32", or "int8", then ...
* If input’s dataType is not "uint8", then ...
* If allowedDataTypes does not contain input’s dataType, then ... _&larr; used for pooling/reduction/unary ops_

So these would be rephrased as:

* If input’s dataType is neither "float32" nor "float16", then ... _&larr; works but fancy?_
* If input’s dataType is neither "float32", "float16", "int32", nor "int8", then ... _&larr; awkward?_
* If input’s dataType is neither "float32" nor "float16" nor "int32" nor "int8", then ... _&larr; awkward?_
* If input’s dataType is not "uint8", then ... _&larr; inconsistent?_
* If allowedDataTypes does not contain input’s dataType, then ...

We could also go for:

* If input’s dataType is not in « "float32", "float16" », then ...
* If input’s dataType is not in « "float32", "float16", "int32", "int8" », then ...
* If input’s dataType is not "uint8", then ... _&larr; inconsistent?_
* If input's dataType is not in allowedDataTypes , then ... 

Preference?

I'll also offer a table-based approach, e.g.

* If input's dataTypes is not an allowed data type for "op" and "input", then ...

... where in an appendix there's a table of op / arg / allowed types.

Reply:
Additional alternatives, trying to work "equal" in there:

* If input’s dataType is not equal to either "float32" or "float16", then ...
* If input’s dataType is not equal to any of "float32", "float16", "int32", or "int8", then ...
* If input’s dataType is not equal to "uint8", then ...
* If allowedDataTypes does not contain input’s dataType, then ... _&larr; used for pooling/reduction/unary ops_

Reply:
Aah, the paradox of choice... I'll accept it as-is and merge, as it sounds like it may well change in the follow-on change anyway, but from the above, I rather like that list approach:

`If input’s dataType is not « "float32", "float16", "int32", "int8" », then ...`